docs/resource_sets.rst

```diff
@@ -0,0 +1,86 @@
+of resources that is independent from other sets (that are not part of the partial compile).
+
+Constraints and rules
+************************
+
+To ensure partial compiles work correctly some constraints and rules where put in place:
+
+* A resource cannot be part of multiple resource sets at the same time.
+* A resource doesn't have to be assigned to a resource set.
+* Resources cannot migrate to a different resource set using a partial compile. This operation requires a full compile.
+* If a partial export contains a resource set, it must be complete, i.e. all its resources should be present in the partial export.
+* A partial compile can never update/remove resources that were not assigned to a specific resource set (adding resources is allowed).
+* The new version of the model that results from a partial compile, should have a dependency graph that is closed within the
+  exported resource sets (i.e. doesn't have any dependencies in other resource sets).
+* A partial compile can update multiple resource sets at the same time.
+
+Exporting a partial model to the server
+******************************************************
+To export a partial model to the server, 2 options can be specified to the ``inmanta export`` command.
+
+- A ``--partial`` option is added to indicate that the model being compiled, only contains the resources that should be updated with respect to the previous version of the model.
+
+- A ``--delete-resource-set <resource-set-name>`` option is added. This option can be passed multiple times and indicates that the resource set with the given name should be removed from the model. This option can only be used in combination with the previous one. Note that it is also possible to implicitly delete resource sets in a partial compile by using an ``std::ResourceSet`` that doesn't contains any resources
+
+Limitations
+*************
+* With partial compiles the compiler cannot verify all constraints that would be checked when a full compile is ran. For example, not all index constraints can be verified. It's the responsibility of the model developer to make sure that these constraints are satisfied.
+* By only doing partial compile it can happen that a shared resource becomes obsolete: several resources that belong to a different resource set can depend on a specific shared resource (not associated with a specific resource set).
+  When a partial compile deletes the last resource that was depending on the shared resource, the shared resource will become obsolete, but it is kept as a resource managed by the server because partial compiles cannot delete shared resources.
+  To delete shared resources a full compile is needed. A solution to this is to perform scheduled full compiles to garbage-collect these shared resources.
+  To schedule full compiles, the :inmanta.environment-settings:setting:`auto_full_compile` environment setting should be used.
```

Comment:
I don't think so, but I believe this is a link to [the documentation for this setting](https://docs.inmanta.com/community/dev/reference/environmentsettings.html#auto_full_compile)

Reply:
I see, that link is like one of those blocks of texts that we discussed yesterday. it does not provide much value for the end-user without an example. If you could provide an example here, that'd be better. Otherwise, we need to see how to tackle this in a nice way.

Reply:
I think in the context of a setting specification the description there is appropriate. But I do agree that in this document we might want to be a bit less formal. I'll add an example.

Reply:
Added a small example, let me know if it doesn't suffice.